modules/docs/pages/replication-protocol.adoc

```diff
@@ -105,22 +105,36 @@ below.
 [[getcheckpoint]]
 getCheckpoint
 ^^^^^^^^^^^^^
-
+`collection` [3.1+]: Indicates the index of the collection to operate on for this message, indexed by the list resolved in `getCollections`.  If legacy `getCheckpoint` is used, this property *must* be omitted +
 `client`: Unique ID of client checkpoint to retrieve
 
 Retrieves a checkpoint stored on the receiver. The checkpoint is a JSON
 object that's stored as the value of the key given by the `client`
-property.
+property.  This property should be derived in a way that is collection
+aware (i.e. each collection can have its own checkpoint).  If this message is sent before `getCollections` then the replication will operate in legacy mode and only the default collection can be replicated. 
 
 Response:
 
 `rev`: The MVCC revision ID of the checkpoint +
 Body: JSON data of the checkpoint
 
+[[getCollections]]
+getCollections [3.1+]
+^^^^^^^^^^^^^^^^^^^^^
+
+Body: JSON dictionary
+
+A collection aware parallelized version of `getCheckpoint`.  If this message is sent before any `getCheckpoint` calls then replication will operate in 3.1+ mode, which can fully handle collections.  The body of this message will contain two arrays (`collections` and `checkpoint_ids`) each of the same length.  The combination of the two at a given index provide information about a checkpoint to retrieve (i.e. request checkpoint X for collection Y).  
+
+Response:
+
+An array of items each identical in structure to the result of `getCheckpoint`.  The entries will be ordered according to the order in the request message.
```

Comment:
Provide info about "null" value for the collection that is not found and empty dict for the collection that doesn't have the remote checkpoint.